Add some user stories to REQUIREMENTS.md (#2)
opencontainers/wg-freebsd-runtime: docs/REQUIREMENTS.md

## Definitions

- **Namespace**: A namespace allows a container to have its own isolated
  instance of a system resource. When using FreeBSD jails, this can be network,
  UTS or IPC.

## User Stories

1. As a user, I want control over which namespaces are private to a container
   and which are shared with the host.

> 2. As a user, I want to be able to group containers into pods which share

Reviewer: ```suggestion
2. As a user, I want to be able to define groups of containers which share
```

nit: pod isn't an OCI term, but is a Kubernetes concept.  Slightly different level of abstraction.

   namespaces.
3. As a user, I want control over resources used by a container, in particular
   CPU, memory and possibly network usage.
4. As a user, I want control over which devices are available inside a
   container, for instance to allow a container access to GPU functions or other
   hardware.
5. As a user, I want control over whether a container can mount filesystems and
   which types of filesystems can be mounted.
6. As a user, I want containers to be able to support nesting, for instance to

>    allow the use of container engines such as Podman or containerd.

Reviewer: nit: containerd itself does not require container nesting

Author: I think I need to reword this - I was thinking about situations when you want to run e.g. containerd inside a container, perhaps for CI purposes. Podman also does not require nesting but can be run nested in some other container (typically one marked as privileged) on Linux.
